crates/move-stackless-bytecode/src/function_target_pipeline.rs

```diff
@@ -409,15 +416,134 @@ impl FunctionTargetsHolder {
             }
         }
 
-        func_env.get_name_str().strip_suffix("_inv").map(|name| {
-            if let Some(struct_env) = func_env
-                .module_env
-                .find_struct(func_env.symbol_pool().make(name))
-            {
-                self.datatype_invs
-                    .insert(struct_env.get_qualified_id(), func_env.get_qualified_id());
+        if let Some(spec_attr) = func_env
+            .get_toplevel_attributes()
+            .get_(&Verification(VerificationAttribute::SpecOnly))
+        {
+            if func_env.get_name_str().contains("type_inv") {
+                return;
+            }
+
+            let env = func_env.module_env.env;
+
+            let inner_attrs = match &spec_attr.value {
+                Attribute_::Parameterized(_, inner_attrs) => inner_attrs,
+                _ => &UniqueMap::new(),
+            };
+
+            let is_inv_target: bool =
+                inner_attrs.contains_key_(&AttributeName_::Unknown(Symbol::from("inv_target")));
+
+            if is_inv_target {
+                let function_spec = inner_attrs
+                    .get_(&AttributeName_::Unknown(Symbol::from("inv_target")))
+                    .unwrap();
+
+                match Self::parse_module_access(function_spec, env) {
+                    Some((module_name, struct_name)) => {
+                        if let Some(module_env) = env.find_module(&module_name) {
```

Comment:
does the compiler validate the module name, such that `find_module` always returns `Some`? If yes, let's use `unwrap` instead. If no, let's handle the `None` case.